tortoise/verifying_algorithm.go

```diff
@@ -11,52 +13,104 @@ import (
 
 // ThreadSafeVerifyingTortoise is a thread safe verifying tortoise wrapper, it just locks all actions.
 type ThreadSafeVerifyingTortoise struct {
-	trtl  *turtle
-	mutex sync.RWMutex
+	trtl      *turtle
+	logger    log.Log
+	lastRerun time.Time
+	mutex     sync.RWMutex
 }
 
 // Config holds the arguments and dependencies to create a verifying tortoise instance.
 type Config struct {
-	LayerSize int
-	Database  blockDataProvider
-	Hdist     uint32
-	Log       log.Log
-	Recovered bool
+	LayerSize       int
+	Database        blockDataProvider
+	ATXDB           atxDataProvider
+	Clock           layerClock
+	Hdist           uint32 // hare lookback distance: the distance over which we use the input vector/hare results
+	Zdist           uint32 // hare result wait distance: the distance over which we're willing to wait for hare results
+	ConfidenceParam uint32 // confidence wait distance: how long we wait for global consensus to be established
+	GlobalThreshold uint8  // threshold required to finalize blocks and layers (0-100)
+	LocalThreshold  uint8  // threshold that determines whether a node votes based on local or global opinion (0-100)
+	WindowSize      uint32 // tortoise sliding window: how many layers we store data for
+	Log             log.Log
+	Recovered       bool
+	RerunInterval   time.Duration // how often to rerun from genesis
```

Comment:
`RerunInterval` is only used in `tortoise/verifying_algorithm.go` but it's still part of `turtle`. is it because we want to persist this setting?

Reply:
yup